numpy/lib/arraysetops.py

```diff
@@ -332,14 +338,45 @@ def intersect1d(ar1, ar2, assume_unique=False):
     >>> from functools import reduce
     >>> reduce(np.intersect1d, ([1, 3, 4, 3], [3, 1, 2, 1], [6, 3, 4, 2]))
     array([3])
+    
+    To return the indices of the values common to the input arrays
+    along with the intersected values:
+    
+    >>> np.intersect1d([1,2,3,4], [2,1,4,6], return_indices=True)
+    (array(1,2,4), array([0,1,3]), array([1,0,2]))
     """
-    if not assume_unique:
+    if not (assume_unique or return_indices):
         # Might be faster than unique( intersect1d( ar1, ar2 ) )?
+        # Added the extra conditional because np.unique uses an additional 
+        # keyword when return_indices is specified for intersect1d. 
         ar1 = unique(ar1)
         ar2 = unique(ar2)
-    aux = np.concatenate((ar1, ar2))
+
+
+    if return_indices:
+      if not assume_unique:
+        a1, ind1 = unique(ar1, return_index = True) #a1 not necessary
+        a2, ind2 = unique(ar2, return_index = True)
```

Comment:
Strange formatting here - we always use multiples of four spaces to indent, and PEP8 says no spaces around keyword arguments

Reply:
It's a relic i forgot to remove, I'll fix this.